src/main/java/org/zalando/nakadi/domain/BatchItem.java

```diff
@@ -64,15 +68,29 @@ public static EmptyInjectionConfiguration build(final int position, final boolea
     private int eventSize;
 
     public BatchItem(
-            final String event,
+            final String rawEvent,
             final boolean strictParsing,
             final EmptyInjectionConfiguration emptyInjectionConfiguration,
             final InjectionConfiguration[] injections,
             final List<Integer> skipCharacters) {
-        this.rawEvent = event;
+        this.rawEvent = rawEvent;
         this.skipCharacters = skipCharacters;
-        this.event = strictParsing ? StrictJsonParser.parseObject(event) : new JSONObject(event);
-        this.eventSize = event.getBytes(StandardCharsets.UTF_8).length;
+        this.event = new JSONObject(rawEvent);
+
+        // We will first release in a shadow mode when the strict parser does second parsing
+        // and we compare if the result is the same as with the old parser; we also log invalid events;
+        if (strictParsing) {
+            try {
+                final JSONObject shadowEvent = StrictJsonParser.parseObject(rawEvent);
+                if (!shadowEvent.toString().equals(this.event.toString())) {
+                    LOG.debug("[STRICT_JSON_DIFF] Strict parser produced different output for event: {}", rawEvent);
```

Comment:
could you please log both parsed events to compare them later?

Reply:
ok, will do

Reply:
https://github.com/zalando/nakadi/pull/882/commits/dc5fe41e80cbfc7ba40e8129b6b407e7ee565973